### ISSUE TITLE
Feat.req. Native VR / OpenXR support

### ISSUE THREAD

**Issue by necros2k7** [rt64/rt64#84]
subj

**Comment by DarioSamo** [comment]
XR support is an interesting task in the long term but definitely not in the immediate roadmap. RT64 can take advantage of features like multiview to provide very good XR performance.

Unless somebody else is interested in picking this up, it's most likely not happening from my side before the other features laid out in the roadmap.

**Comment by DarioSamo** [comment]
> Multiview as in Single Pass Stereo's implementation? https://developer.nvidia.com/vrworks/graphics/singlepassstereo https://developer.nvidia.com/downloads/vrworks/secure/3.6.2/public_3.6_06112024.zip

Yes, although you don't need NVIDIA's library for it.

**Comment by DarioSamo** [comment]
> Wait, so you mean you just borrowed the MV/SPS code from NVIDIA VRWorks it did RT64 devs write their own implementation? If so, is it available anywhere? I've been suggesting SPS for a couple projects to improve performance, but something that's not NVIDIA hardware-locked would probably be a better idea.

No, Multiview is just part of modern APIs. You don't need NVIDIA's solution for it. I didn't write anything to support it yet, but it's available as an extension in Vulkan and a native D3D12 feature.

https://registry.khronos.org/vulkan/specs/latest/man/html/VK_KHR_multiview.html
